src/managers/NAVManager.sol

```diff
@@ -0,0 +1,302 @@
+    /// @inheritdoc INAVManager
+    function setNAVHook(INAVHook navHook_) external onlyHubManager {
+        navHook = navHook_;
+        emit SetNavHook(address(navHook_));
+    }
+
+    /// @inheritdoc INAVManager
+    function updateManager(address manager_, bool canManage) external onlyHubManager {
+        require(manager_ != address(0), EmptyAddress());
+
+        manager[manager_] = canManage;
+
+        emit UpdateManager(manager_, canManage);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Account creation
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc INAVManager
+    function initializeNetwork(uint16 centrifugeId) external onlyManager {
+        require(accountCounter[centrifugeId] == 0, AlreadyInitialized());
+
+        hub.createAccount(poolId, equityAccount(centrifugeId), false);
+        hub.createAccount(poolId, liabilityAccount(centrifugeId), false);
+        hub.createAccount(poolId, gainAccount(centrifugeId), false);
+        hub.createAccount(poolId, lossAccount(centrifugeId), false);
+
+        accountCounter[centrifugeId] = 5;
+
+        emit InitializeNetwork(centrifugeId);
+    }
+
+    /// @inheritdoc INAVManager
+    function initializeHolding(ShareClassId scId, AssetId assetId, IValuation valuation) external onlyManager {
+        uint16 centrifugeId = assetId.centrifugeId();
+        uint16 index = accountCounter[centrifugeId];
+        require(index > 0, NotInitialized());
+        require(index < type(uint16).max, ExceedsMaxAccounts());
+
+        AccountId assetAccount_ = assetIdToAccountId[centrifugeId][assetId];
+        if (assetAccount_.isNull()) {
+            assetAccount_ = withCentrifugeId(centrifugeId, index);
+            assetIdToAccountId[centrifugeId][assetId] = assetAccount_;
+        }
+
+        hub.createAccount(poolId, assetAccount_, true);
+        hub.initializeHolding(
+            poolId,
+            scId,
+            assetId,
+            valuation,
+            assetAccount_,
+            equityAccount(centrifugeId),
+            gainAccount(centrifugeId),
+            lossAccount(centrifugeId)
+        );
+
+        accountCounter[centrifugeId] = index + 1;
+
+        emit InitializeHolding(scId, assetId);
+    }
+
+    /// @inheritdoc INAVManager
+    function initializeLiability(ShareClassId scId, AssetId assetId, IValuation valuation) external onlyManager {
+        uint16 centrifugeId = assetId.centrifugeId();
+        uint16 index = accountCounter[centrifugeId];
+        require(index > 0, NotInitialized());
+        require(index < type(uint16).max, ExceedsMaxAccounts());
+
+        AccountId expenseAccount_ = assetIdToAccountId[centrifugeId][assetId];
+        if (expenseAccount_.isNull()) {
+            expenseAccount_ = withCentrifugeId(centrifugeId, index);
+            assetIdToAccountId[centrifugeId][assetId] = expenseAccount_;
```

Comment:
Because `assetIdToAccountId` is used in both `initializeHolding` and `initializeLiability`, couldn't the manager mess up things? Should we use two different storage instead?